runtime/src/comm/ugni/comm-ugni.c

```diff
@@ -18,6 +18,12 @@
  * limitations under the License.
  */
 
+#ifdef __STRICT_ANSI__
+// For builds with NVIDIA we define this, but this source is not compatible with
+// it. So, undefine if it was defined
+#undef __STRICT_ANSI__
+#endif
+
```

Comment:
How bad of a change is it to adjust the implementation to use ansi compatible code?

Reply:
I have a feeling that the issues might be rooted in headers, likely ugni headers that are not ours. Especially one about `pthread_sigmask` is a bit interesting/worrisome.

```
comm-ugni.c:640:25: error: tentative definition of variable with internal linkage has incomplete non-array type 'struct sigaction' [-Werror,-Wtentative-definition-incomplete-type]
  640 | static struct sigaction previous_SIGBUS_sigact;
      |                         ^
comm-ugni.c:640:15: note: forward declaration of 'struct sigaction'
  640 | static struct sigaction previous_SIGBUS_sigact;
      |               ^
comm-ugni.c:1395:38: error: unknown type name 'siginfo_t'
 1395 | static void      SIGBUS_handler(int, siginfo_t *, void *);
      |                                      ^
comm-ugni.c:3189:20: error: variable has incomplete type 'struct sigaction'
 3189 |   struct sigaction act;
      |                    ^
comm-ugni.c:640:15: note: forward declaration of 'struct sigaction'
  640 | static struct sigaction previous_SIGBUS_sigact;
      |               ^
comm-ugni.c:3191:18: error: use of undeclared identifier 'SA_RESTART'
 3191 |   act.sa_flags = SA_RESTART | SA_SIGINFO;
      |                  ^
comm-ugni.c:3191:31: error: use of undeclared identifier 'SA_SIGINFO'
 3191 |   act.sa_flags = SA_RESTART | SA_SIGINFO;
      |                               ^
comm-ugni.c:3192:7: error: call to undeclared function 'sigemptyset'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
 3192 |   if (sigemptyset(&act.sa_mask) != 0) {
      |       ^
comm-ugni.c:3199:7: error: call to undeclared function 'sigaction'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
 3199 |   if (sigaction(SIGBUS, &act, &previous_SIGBUS_sigact) != 0) {
      |       ^
comm-ugni.c:3209:32: error: unknown type name 'siginfo_t'
 3209 | void SIGBUS_handler(int signo, siginfo_t *info, void *context)
      |                                ^
comm-ugni.c:3227:24: error: use of undeclared identifier 'BUS_ADRERR'
 3227 |   if (info->si_code == BUS_ADRERR) {
      |                        ^
comm-ugni.c:3345:7: error: call to undeclared function 'sigaction'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
 3345 |   if (sigaction(SIGBUS, &previous_SIGBUS_sigact, NULL) != 0) {
      |       ^
comm-ugni.c:3357:9: error: call to undeclared function 'sigemptyset'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
 3357 |     if (sigemptyset(&sigbus_set) != 0
      |         ^
comm-ugni.c:3358:12: error: call to undeclared function 'sigaddset'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
 3358 |         || sigaddset(&sigbus_set, SIGBUS) != 0
      |            ^
comm-ugni.c:3359:12: error: call to undeclared function 'pthread_sigmask'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
 3359 |         || pthread_sigmask(SIG_UNBLOCK, &sigbus_set, NULL) != 0
      |            ^
comm-ugni.c:3359:28: error: use of undeclared identifier 'SIG_UNBLOCK'
 3359 |         || pthread_sigmask(SIG_UNBLOCK, &sigbus_set, NULL) != 0
      |                            ^
comm-ugni.c:3361:28: error: use of undeclared identifier 'SIG_BLOCK'
 3361 |         || pthread_sigmask(SIG_BLOCK, &sigbus_set, NULL) != 0) {
      |                            ^
comm-ugni.c:640:25: error: tentative definition has type 'struct sigaction' that is never completed
  640 | static struct sigaction previous_SIGBUS_sigact;
      |                         ^
comm-ugni.c:640:15: note: forward declaration of 'struct sigaction'
  640 | static struct sigaction previous_SIGBUS_sigact;
      |
```

Reply:
If you prefer, I can fire up a follow up issue to revert my band-aid and make this file ANSI-compliant. I don't know if it'd worth the effort, though.

Reply:
Based on the amount of errors, that doesn't look worth it for this PR. Just turning of `__STRICT_ANSI__` makes me a little nervous, but like you said this may be out of our control.

It is probably worth an issue though

Reply:
I agree. If it helps, though, as far as I can see on a quick look, NVIDIA runtime build is the only time we require `__STRICT_ANSI__`. The most worrisome part of this for me is (maybe you, too) when/if we start enforcing ANSI compliance in the runtime as a whole, and this file flies under the radar for that change.

Reply:
> The most worrisome part of this for me is (maybe you, too) when/if we start enforcing ANSI compliance in the runtime as a whole, and this file flies under the radar for that change.

That was exactly my concern.

And yes, `-D__STRICT_ANSI__` is only for nvidia codegen, I know because I added it :smile:. Although grepping through the code shows that there is some old cygwin stuff turning off `__STRICT_ANSI__` that is 17+ years old